roles/couchbase-infra/tasks/main.yml

```diff
@@ -0,0 +1,209 @@
+---
+# Install Couchbase Operator (Assume rook is already in place? or should we check?)
+- name: Create CRD for Couchbase Operator
+  k8s:
+    definition: "{{ lookup('file', 'crd.yaml') }}"
+  when: couchbase.servers.size > 0
+
+- name: Reticulating Splines
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        # This is the namespace coded into the operator
+        # from operatorhub.io. Best just to embrace it
+        # rather than fight it.
+        name: my-couchbase-enterprise
+  when: couchbase.servers.size > 0
+
+- name: Create Cluster Role for Service Account
+  k8s:
+    definition: "{{ lookup('file', 'cluster-role-sa.yaml') }}"
+  when: couchbase.servers.size > 0
+
+- name: Create Service Account
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: ServiceAccount
+      metadata:
+        name: couchbase-operator
+        #namespace: '{{ meta.namespace }}'
+        namespace: my-couchbase-enterprise
+  when: couchbase.servers.size > 0
+
+- name: Create ClusterRoleBinding for the Service Account
+  k8s:
+    definition:
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRoleBinding
+      metadata:
+        name: couchbase-operator-rolebinding
+      roleRef:
+        apiGroup: rbac.authorization.k8s.io
+        kind: ClusterRole
+        name: couchbase-operator
+      subjects:
+      - kind: ServiceAccount
+        name: couchbase-operator
+        #namespace: '{{ meta.namespace }}'
+        namespace: my-couchbase-enterprise
+  when: couchbase.servers.size > 0
+
+- name: Create Cluster Role for the user
+  k8s:
+    definition: "{{ lookup('file', 'cluster-role-user.yaml') }}"
+  when: couchbase.servers.size > 0
+
+- name: Create ClusterRoleBinding for the User
+  k8s:
+    definition:
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRoleBinding
+      metadata:
+        name: couchbasecluster-rolebinding
+      roleRef:
+        apiGroup: rbac.authorization.k8s.io
+        kind: ClusterRole
+        name: couchbasecluster
+  when: couchbase.servers.size > 0
+
+- name: Create ClusterRole for the Benchmark User
+  k8s:
+    definition:
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRole
+      metadata:
+        name: benchmark-operator-for-cb
+        namespace: '{{ meta.namespace }}'
+      rules:
+      - apiGroups:
+        - ""
+        resources:
+        - namespaces
+        verbs:
+        - get
+      - apiGroups:
+        - couchbase.com
+        resources:
+        - couchbaseclusters
+        verbs:
+        - '*'
+  when: couchbase.servers.size > 0
+
+- name: Create ClusterRoleBinding for the Benchmark User
```

Comment:
I thought part of the readme was to import this role prior to running?

Reply:
I had order-of-operation problems initially when developing, but I see now I can merge everything into the code. I will get rid of the files in deploy/.